src/objects.js

```diff
@@ -13491,3 +13491,497 @@ ReplayControls.prototype.getInverseEvent = function(event) {
+        }
+        const temp = element[0].childNodes[0].nodeValue;
+        return temp.charAt(0).toUpperCase() + temp.slice(1);
+    }
+
+    this.initDotted = function () {
+        const element = this.src.getElementsByTagName("dot");
+        return element.length != 0;
+    }
+
+    this.pitch = this.initPitch();
+    this.type = this.initType();
+    this.dotted = this.initDotted();
+}
+
+/**
+ * @description - Creates and instance of an object representing a part in a musical work.
+ * @param {Element} src - The contents of a <part> tag in a .musicxml file. 
+ * @param {String} name - The name of the part.
+ */
+function Part(src, name="untitled") {
+    this.src = src;
+    this.notes = [];
+    this.name = name;
+    const xmlMeasures = this.src.getElementsByTagName("measure");
+    for (let i = 0; i < xmlMeasures.length; ++i) {
+        const xmlNotes = xmlMeasures[i].getElementsByTagName("note");
+        for (let j = 0; j < xmlNotes.length; ++j) this.notes.push(new Note(xmlNotes[j]));
+    }
+}
+
+/**
+ * @description - Creates a MusicReader object that parses MusicXML files.
+ * @constructor
+ * @param {String} src - The contents of a .musicxml file.
+ * @param {String} name - The name of the MusicReader.
+ */
+function MusicReader(src, name = "untitled") {
+    this.src = src;
+    this.name = name;
+    this.parts = [];
+    const parser = new DOMParser();
+    const xmlDoc = parser.parseFromString(this.src, "text/xml");
+    const parts = xmlDoc.getElementsByTagName("part");
+    const scoreParts = xmlDoc.getElementsByTagName("part-name");
+    for (let i = 0; i < parts.length; ++i) {
+        this.parts.push(new Part(parts[i], scoreParts[i].textContent));
+    }
+}
+
+/**
+ * @description - Creates a MusicWriter object with the data from a MusicReader.
+ * @constructor
+ * @param {MusicReader} musicReader - The contents of a MusicReader.
+ */
+function MusicWriter(musicReader) {
+    const DEFAULT_TEMP0 = 60;
+    this.mMusicReader = musicReader;
+    this.myRoot = new Document();
+}
+
+/**
+ * @description - Creates a .xml file to pass back to the user.
+ * @returns {String} - A String holding the contents of the .xml file.
+ */
+MusicWriter.prototype.getFile = function () {
+    this.initFile()
+    return new XMLSerializer().serializeToString(this.myRoot.documentElement);
+}
+
+/**
+ * @description - This function starts the process of converting a Music XML file into a usable 
+ *                NetsBlox project.
+ */
+MusicWriter.prototype.initFile = function () {
+    const room = this.createRoom();
+    this.myRoot.appendChild(room);
+}
+
+/**
+ * @description - This function creates the content of a <room> tag in a NetsBlox project.
+ * @returns {Element} - The <room> tag.
+ */
+MusicWriter.prototype.createRoom = function () {
+    const room = this.myRoot.createElement("room");
+    room.setAttribute("name", this.mMusicReader.name);
+    room.setAttribute("app", "NetsBlox 2.3.2, http://netsblox.org");
```

Comment:
This shouldn't be hard-coded and should use the current version (defined in store-ext.js).